src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -32,7 +32,7 @@ public class InputTagHelper : TagHelper
                 { "Url", "url" },
                 { "EmailAddress", "email" },
                 { "Date", "date" },
-                { "DateTime", "datetime" },
+                { "DateTime", "datetime-local" },
```

Comment:
That is, @Eilon please confirm your intent. Obviously my leaning was toward changing only case (1) because (2) seems more like an explicit ask for `type="datetime"`. But, I can see (2) is not _that_ much more explicit.